crates/nu-command/src/commands/filters/select.rs

```diff
@@ -154,3 +162,65 @@ fn select(args: CommandArgs) -> Result<OutputStream, ShellError> {
     }))
     .into_output_stream())
 }
+
+#[cfg(test)]
+mod tests {
+    use nu_protocol::ColumnPath;
+    use nu_source::Span;
+    use nu_source::SpannedItem;
+    use nu_source::Tag;
+    use nu_stream::InputStream;
+    use nu_test_support::value::nothing;
+    use nu_test_support::value::row;
+    use nu_test_support::value::string;
+
+    use super::select;
+    use super::Command;
+    use super::ShellError;
+
+    #[test]
+    fn examples_work_as_expected() -> Result<(), ShellError> {
+        use crate::examples::test as test_examples;
+
+        test_examples(Command {})
+    }
+
+    #[test]
+    fn select_using_sparse_table() {
+        // Create a sparse table with 3 rows:
+        //   col_foo | col_bar
+        //   -----------------
+        //   foo     |
+        //           | bar
+        //   foo     |
+        let input = vec![
+            row(indexmap! {"col_foo".into() => string("foo")}),
+            row(indexmap! {"col_bar".into() => string("bar")}),
+            row(indexmap! {"col_foo".into() => string("foo")}),
+        ];
+
+        let expected = vec![
+            row(
+                indexmap! {"col_none".into() => nothing(), "col_foo".into() => string("foo"), "col_bar".into() => nothing()},
+            ),
+            row(
+                indexmap! {"col_none".into() => nothing(), "col_foo".into() => nothing(), "col_bar".into() => string("bar")},
+            ),
+            row(
+                indexmap! {"col_none".into() => nothing(), "col_foo".into() => string("foo"), "col_bar".into() => nothing()},
+            ),
+        ];
+
+        let actual = select(
+            Tag::unknown(),
+            vec![
+                ColumnPath::build(&"col_none".to_string().spanned(Span::unknown())),
```

Comment:
I'm actually wondering whether `select` should fail if the column isn't present in any of the rows, e.g. `select col_none` would fail here. It could prevent typos and injecting new empty columns.